syncr_backend/run_backend.py

```diff
@@ -1,31 +1,79 @@
 #!/usr/bin/env python
 import argparse
+import sys
+import threading
+from typing import Any
 from typing import List
 
-from syncr_backend import drop_init
-from syncr_backend import node_init
+from syncr_backend.init import drop_init
+from syncr_backend.init import node_init
+from syncr_backend.network import send_requests
+from syncr_backend.network.listen_requests import listen_requests
 
 
-def execute_node_function(function_name: str, args: List[str]):
+def run_backend() -> None:
     """
-    Runs a function with the given args
+    Runs the backend
+    """
+    input_args_parser = argparse.ArgumentParser()
+    input_args_parser.add_argument(
+        "ip",
+        type=str,
+        nargs=1,
+        help='ip to bind listening server',
+    )
+    input_args_parser.add_argument(
+        "port",
+        type=str,
+        nargs=1,
+        help='ip to bind listening server',
+    )
+    arguments = input_args_parser.parse_args()
 
-    :param function_name: string name of the function to run
-    :param args: arguments for the function to run
+    request_listen_thread = threading.Thread(
+        target=listen_requests,
+        args=[arguments.ip[0], arguments.port[0]],
+    )
+    request_listen_thread.start()
+    read_cmds_from_cmdline()
+    request_listen_thread.join()
+
+
+def read_cmds_from_cmdline() -> None:
     """
+    Read and execute commands given as cmdline input
+    """
+    exit = False
+    while not exit:
+        command = input("5yncr >>> ").split(' ')
+        return_list = ["", []]
+        parse_cmd_thread = threading.Thread(
+            target=parse_cmd,
+            args=[command, return_list],
+        )
+        parse_cmd_thread.start()
```

Comment:
Why do thread.start followed by thread.join?

Reply:
In a future commit, we will have a more elegant way to exit the program. Should an exit signal be sent to the command thread and the backend thead, we do not want to have the command thread read it first and exit then terminating the program while the child process is alive. The thread.join is to ensure that both threads come to an end before the program ends.